apps/keyboard/js/keyboard/target_handlers_manager.js

```diff
@@ -107,7 +107,7 @@ TargetHandlersManager.prototype._callTargetAction = function(action,
   }
 
   this.promiseQueue = this.promiseQueue.then(function() {
-    handler[action](press);
+    return handler[action](press);
```

Comment:
Ouch. Could you do the 319MB performance test again? It looks like we do not wait for API round trip in the previous patch.